datalad/metadata/find_datasets.py

```diff
@@ -0,0 +1,117 @@
+# emacs: -*- mode: python; py-indent-offset: 4; tab-width: 4; indent-tabs-mode: nil -*-
+# ex: set sts=4 ts=4 sw=4 noet:
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+#
+#   See COPYING file distributed along with the datalad package for the
+#   copyright and license terms.
+#
+# ## ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ### ##
+"""Interface for managing metadata
+"""
+
+__docformat__ = 'restructuredtext'
+
+
+import re
+from datalad.interface.base import Interface
+from datalad.distribution.dataset import datasetmethod, EnsureDataset, \
+    require_dataset
+from ..support.param import Parameter
+from ..support.constraints import EnsureNone
+from ..log import lgr
+from . import get_metadata, flatten_metadata_graph
+
+
+class FindDatasets(Interface):
+    """
+    """
+
+    _params_ = dict(
+        dataset=Parameter(
+            args=("-d", "--dataset"),
+            doc="""specify the dataset to perform the query operation on. If
+            no dataset is given, an attempt is made to identify the dataset
+            based on the current working directory and/or the `path` given""",
+            constraints=EnsureDataset() | EnsureNone()),
+        match=Parameter(
+            args=("match",),
+            metavar='REGEX',
+            doc="expression to match against all meta data values"),
+        #match=Parameter(
+        #    args=('-m', '--match',),
+        #    metavar='REGEX',
+        #    action='append',
+        #    nargs=2,
+        #    doc="""Pair of two regular expressions to match a property and its
+        #    value.[CMD:  This option can be given multiple times CMD]"""),
+        report=Parameter(
+            args=('-r', '--report'),
+            metavar='PROPERTY',
+            action='append',
+            # could also be regex
+            doc="""name of the property to report for any match.[CMD:  This
+            option can be given multiple times. CMD] If none are given, all
+            properties are reported."""),
+    )
+
+    @staticmethod
+    @datasetmethod(name='query')
```

Comment:
I would still then have just 'search' (or find, but my soul is rotten with apt), and what is output be dictated by options (if needed), not separate long commands.

On August 16, 2016 3:39:27 AM EDT, Michael Hanke <notifications@github.com> wrote:
>> +        #    action='append',
>> +        #    nargs=2,
>> +        #    doc="""Pair of two regular expressions to match a
>property and its
>> +        #    value.[CMD:  This option can be given multiple times
>CMD]"""),
>> +        report=Parameter(
>> +            args=('-r', '--report'),
>> +            metavar='PROPERTY',
>> +            action='append',
>> +            # could also be regex
>> +            doc="""name of the property to report for any
>match.[CMD:  This
>> +            option can be given multiple times. CMD] If none are
>given, all
>> +            properties are reported."""),
>> +    )
>> +
>> +    @staticmethod
>> +    @datasetmethod(name='query')
>
>Good catch. "query" here is a left-over. I think we should not attempt
>a proper `query` command without using a real query language/"DB"
>combo. This implementation simplifies things to a manageable degree (I
>hope) and can find datasets. Hence the name choice.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/datalad/datalad/pull/682/files/f7adc3bee6ab0ea6518a5466c431500d5dac39e4#r74888592